### ISSUE TITLE
nx-go not working with modules/sub-packages (cannot find package ...)

### ISSUE THREAD

**Issue by stefanbinder** [nx-go/nx-go#19]
Hi,

I'm trying to create a nx-go-module, but then I'm not able to run it with the `nx r go-module:serve` command.
But running it with `go run main.go` is working without any problem.

I was creating an nx-go application:
`nx g @nx-go/nx-go:application web-1 --directory=bff`

go.mod:
```
module github.com/myuser/myrepo/bff-web-1

go 1.15
```

main.go:
```
package main

import (
  "fmt"
  "github.com/myuser/myrepo/bff-web-1/submod"
)

func Hello(name string) string {
  mySubMod := submod.SubMod{
    ID: "1234",
  }
	result := "Hello " + name + " " + mySubMod.ID
	return result
}

func main() {
	fmt.Println(Hello("bff-web-1"))
}

```

submod/submod.go:
```
package submod

type SubMod struct {
  ID string
}
```

When I'm running then the module with `nx r bff-web-1:serve` I just get an error:  
```
Executing command: go run apps/bff/web-1/src/main.go
apps/bff/web-1/src/main.go:5:3: cannot find package "github.com/myuser/myrepo/bff-web-1/submod" in any of:
	/usr/local/Cellar/go/1.15.4/libexec/src/github.com/myuser/myrepo/bff-web-1/submod (from $GOROOT)
	/Users/bins/go/src/github.com/myuser/myrepo/bff-web-1/submod (from $GOPATH)
```

My `$GOPATH` is set to: `/Users/MYUSER/go` and added to `$PATH`
Running that module with `go run main.go` it works fine and prints out: `Hello bff-web-1 1234`

I'm kind of new to go's modules, but until now I haven't had problems importing my sub-packages.

**Comment by stefanbinder** [comment]
I found your issue. I've tried to use the `cwd` option, but it was not working because you aren't passing it to the `runGoCommand`.

with: `return runGoCommand(context, 'run', [mainFile], { cmd: options.cmd, cwd: options.cwd })` it is working for me.

**Comment by beeman** [comment]
Hey @stefanbinder, I pushed few fixes and I think this might now work as expected. The `cwd` property in the `runGoCommand` is now set to the project root.

The plugin will now also create a `go.mod` on the top-level if it doesn't exist. I did not succeed having several go.mod files, only a single one. 

Closing this for now, let me know if you still run into any issues - happy to re-open and investigate!
